libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,295 @@
+    angles1 = {-45 + angles1[1],angles1[2],angles1[3]}
+
+    local ans2 = body_forward_kinematics(endpoints2[1]+gait_pos_x[2], endpoints2[2]+gait_pos_y[2], endpoints2[3]+gait_pos_z[2], FRAME_LEN/2, -FRAME_WIDTH/2,gait_rot_z[2])
+    local angles2 = leg_inverse_kinematics(endpoints2[1]+ans2[1]+gait_pos_x[2],endpoints2[2]+ans2[2]+gait_pos_y[2], endpoints2[3]+ans2[3]+gait_pos_z[2])
+    angles2 = {-135 + angles2[1],angles2[2],angles2[3]}
+
+    local ans3 = body_forward_kinematics(endpoints3[1]+gait_pos_x[3], endpoints3[2]+gait_pos_y[3], endpoints3[3]+gait_pos_z[3], -FRAME_LEN/2, -FRAME_WIDTH/2,gait_rot_z[3])
+    local angles3 = leg_inverse_kinematics(endpoints3[1]+ans3[1]+gait_pos_x[3],endpoints3[2]+ans3[2]+gait_pos_y[3], endpoints3[3]+ans3[3]+gait_pos_z[3])
+    angles3 = {135 + angles3[1],angles3[2],angles3[3]}
+
+    local ans4 = body_forward_kinematics(endpoints4[1]+gait_pos_x[4], endpoints4[2]+gait_pos_y[4], endpoints4[3]+gait_pos_z[4], -FRAME_LEN/2, FRAME_WIDTH/2,gait_rot_z[4])
+    local angles4 = leg_inverse_kinematics(endpoints4[1]+ans4[1]+gait_pos_x[4],endpoints4[2]+ans4[2]+gait_pos_y[4], endpoints4[3]+ans4[3]+gait_pos_z[4])
+    angles4 = {45 + angles4[1],angles4[2],angles4[3]}
+    Gaitselect()
+    current = {angles1[1],angles1[2],angles1[3],angles2[1],angles2[2],angles2[3],angles3[1],angles3[2],angles3[3],angles4[1],angles4[2],angles4[3]}
+
+    if servo_estimate(start_time, current, last_angle) then
+        start_time = millis()
+        calc_gait_sequence()
+        last_angle = current
+    end
+
+    return angles1, angles4, angles3, angles2
+end
+
+-- servo angles when robot is disarmed and resting body on the ground
+local rest_angles = { 45, -90, 40,      -- front right leg (coxa, femur, tibia)
+                     -45, -90, 40,      -- front left leg (coxa, femur, tibia)
+                      45, -90, 40,      -- back right leg (coxa, femur, tibia)
+                     -45, -90, 40}      -- back left leg (coxa, femur, tibia)
+
+local servo_direction = { 1,  1,  1,    -- front right leg (coxa, femur, tibia)
+                         -1, -1, -1,    -- front left leg (coxa, femur, tibia)
+                         -1, -1,  1,    -- back right leg (coxa, femur, tibia)
+                          1,  1, -1}    -- back left leg (coxa, femur, tibia)
+
+function update()
+    x_speed = vehicle:get_control_output(control_input_throttle) * xy_speed_max
+    yaw_speed = vehicle:get_control_output(control_input_yaw) * yaw_speed_max
+
+    body_rot_x = -vehicle:get_control_output(control_input_roll) * body_rot_max
+    body_rot_y = -vehicle:get_control_output(control_input_pitch) * body_rot_max
+
+    if arming:is_armed() then
+        FR_angles, BL_angles, BR_angles, FL_angles = main_inverse_kinematics()
+        angles = {FR_angles[1], FR_angles[2], FR_angles[3], FL_angles[1], FL_angles[2], FL_angles[3], BR_angles[1], BR_angles[2], BR_angles[3], BL_angles[1], BL_angles[2], BL_angles[3]}
```

Comment:
How is `angles` any different then what is already contained in `current` at this point?

Reply:
ok, so main_inverse_kinematics has been updated to both set current and then return it which is a bit odd.